catroid/src/main/java/org/catrobat/catroid/embroidery/DSTFileGenerator.java

```diff
@@ -0,0 +1,295 @@
+			if (minX > stitch.point.x) {
+				minX = stitch.point.x;
+			}
+			if (maxX < stitch.point.x) {
+				maxX = stitch.point.x;
+			}
+			if (minY > stitch.point.y) {
+				minY = stitch.point.y;
+			}
+			if (maxY < stitch.point.y) {
+				maxY = stitch.point.y;
+			}
+		}
+		return new int[] {(int) maxX, (int) minX, (int) maxY, (int) minY };
+	}
+
+	private String getHeaderString(int size, int colorChanges, int[] boundingBox, PointF firstStitch, PointF
+			lastStitch) {
+		final int mx = 0;
+		final int my = 0;
+		final String pd = "*****";
+		StringBuilder stringBuilder = new StringBuilder();
+
+		stringBuilder.append(String.format(DST_HEADER_LABEL,
+				ProjectManager.getInstance().getCurrentProject().getName()))
+				.append(String.format(Locale.getDefault(), DST_HEADER, size, colorChanges, boundingBox[0],
+				boundingBox[1], boundingBox[2], boundingBox[3], (int) (lastStitch.x - firstStitch.x),
+				(int) (lastStitch.y - firstStitch.y), mx, my, pd).replace(' ', '\0'))
+				.append(HEADER_FILL);
+
+		return stringBuilder.toString();
+	}
+
+	@VisibleForTesting
+	public byte[] getHeaderBytes(ArrayList<StitchPoint> stitchPoints) {
+		final int colorChanges = 1;
+		PointF firstStitch = stitchPoints.get(0).point;
+		PointF lastStitch = stitchPoints.get(stitchPoints.size() - 1).point;
+		int[] boundingBox = getBoundingBox(stitchPoints);
+
+		return getHeaderString(stitchPoints.size(), colorChanges, boundingBox, firstStitch, lastStitch).getBytes();
+	}
+
+	private byte[] getBytesForStitchPoint(int x, int y, boolean jump) {
+		char yPart = (char) ((((y & 0x1) << 3) | ((y & 0x2) << 1) | ((y & 0x10) >>> 3)
+				| ((y & 0x20) >>> 5)) << 4);
+		char xPart = (char) (((x >>> 2) & 0xC) | (x & 0x3));
+		char byte0 = (char) (yPart | xPart);
+
+		yPart = (char) ((((y & 0x4) << 1) | ((y & 0x8) >>> 1) | ((y & 0x40) >>> 5)
+				| ((y & 0x80) >>> 7)) << 4);
+		xPart = (char) (((x >>> 4) & 0xC) | ((x >>> 2) & 0x3));
+		char byte1 = (char) (yPart | xPart);
+
+		yPart = (char) (((y >>> 5) & 0x10) | ((y >>> 3) & 0x20));
+		xPart = (char) ((x >>> 6) & 0xC);
+		char byte2 = (char) (yPart | xPart | 0x03);
+
+		if (jump) {
+			byte2 = (char) (byte2 | (0x1 << 7));
+		}
+
+		return new byte[] {(byte) byte0, (byte) byte1, (byte) byte2};
+	}
+
+	@VisibleForTesting
+	public byte[] getStitchesBytes(ArrayList<StitchPoint> stitchPoints) {
+		int xChange;
+		int yChange;
+		StitchPoint currentPoint = stitchPoints.get(0);
+		ByteArray stitchBytes = new ByteArray();
+
+		for (int i = 0; i < stitchPoints.size(); i++) {
+			StitchPoint nextPoint = stitchPoints.get(i);
+			xChange = (int) (nextPoint.point.x - currentPoint.point.x);
+			yChange = (int) (nextPoint.point.y - currentPoint.point.y);
+			int xValue = (xChange < 0 ? CONVERSIONTABLE[(xChange * (-1)) + 121] : CONVERSIONTABLE[xChange]);
+			int yValue = (yChange < 0 ? CONVERSIONTABLE[(yChange * (-1)) + 121] : CONVERSIONTABLE[yChange]);
+			stitchBytes.addAll(getBytesForStitchPoint(xValue, yValue, currentPoint.jump));
```

Comment:
I think it would make sense to move getBytesForStitchPoint into the StitchPoint class and rename it accordingly (getBytes).

The calculation of a stitchpoint with xValue, yValue and currentPoint.jump could be a method in the StitchPoint class as well. Then you could have something pleasant like currentPoint.delta(nextPoint).getBytes() (I would use a name that describes what is happening here better than delta, but I don't fully understand what is going).